### PR TITLE
add documentation for #192

### PR DESCRIPTION
**Pull request by k-okada**
this includes #192

- verbs: adding shell verbs 'cd' and 'source'

### CODE REVIEW

File: docs/index.rst [hunk 1]
```diff
@@ -67,6 +68,12 @@ Each of the following verbs is built-in to the ``catkin`` command and has its ow
 - :doc:`list -- Find and list information about catkin packages in a workspace <verbs/catkin_list>`
 - :doc:`profile -- Manage different named configuration profiles <verbs/catkin_profile>`
 
+Shell support in ``catkin`` command
+-----------------------------------
+
+A bash/zsh user can use wrapper wrapper function, please read :doc:`development/catkin_shell_verbs`.
```
Comment (wjwwood) [inline docs/index.rst]: "wrapper wrapper" looks like a typo.

File: docs/advanced/catkin_shell_verbs.rst [hunk 2]
```diff
@@ -0,0 +1,36 @@
+Shell support in ``catkin`` command
+===================================
+
+When you source following file, you can use bash/zsh wrapper function for shell support.
```
Comment (wjwwood) [inline docs/advanced/catkin_shell_verbs.rst]: "When you ``source`` the following file, you can use bash/zsh shell functions which provide added utility."?

File: docs/advanced/catkin_shell_verbs.rst [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+Change to package directory in source space with `cd` verb.
+
+.. code-block:: text
+
+    usage: catkin cd [ARGS...]
+
+    ARGS are any valid catkin locate arguments
+
+The `source` verb sources the develspace of the containing workspace
```
Comment (wjwwood) [inline docs/advanced/catkin_shell_verbs.rst]: Is it only the devel space or also the install space when appropriate?

**Comment by wjwwood** [comment]
Thanks for working on this @k-okada.

**Comment by k-okada** [comment]
thanks for review, updated docs

File: docs/advanced/catkin_shell_verbs.rst [hunk 4]
```diff
@@ -0,0 +1,36 @@
+Shell support in ``catkin`` command
+===================================
+
+When you source the following file, you can use bash/zsh shell functions which provide added utility.
+
+.. code-block:: shell
+
+    . /opt/ros/indigo/etc/bash_completion.d/catkin_shell_verbs.{bash,zsh}
```
Comment (wjwwood) [inline docs/advanced/catkin_shell_verbs.rst]: Is that the right directory?

Comment (k-okada) [inline docs/advanced/catkin_shell_verbs.rst]: Sorry, it is not correct ,and I think it is not installed, how can I configure setup.py to install ./scripts/catkin_shell_verbs.bash

Comment (wjwwood) [inline docs/advanced/catkin_shell_verbs.rst]: I'm not sure what the right thing to do here is, maybe @jbohren can advise?

You could do something cute like this:

```
$ source `catkin --locate-extra-shell-verbs`
```

And then install the shell scripts that need to be sourced in with the Python code and provide an option to expose their final location.

Comment (k-okada) [inline docs/advanced/catkin_shell_verbs.rst]: ok, how about this 25b8c4fc3fd3cd0107f97bbd45bf0dfb775c226b, I'm not sure
how to get installed location ...
http://stackoverflow.com/questions/50499/how-do-i-get-the-path-and-name-of-the-file-that-is-currently-executing

--
◉ Kei Okada

Comment (NikolausDemmel) [inline docs/advanced/catkin_shell_verbs.rst]: Possibly [`resource_filename`](https://pythonhosted.org/setuptools/pkg_resources.html#resource-extraction) is what you are looking for.

Comment (k-okada) [inline docs/advanced/catkin_shell_verbs.rst]: thanks, in this case, we should install `catkin_shell_verbs.bash` under `lib/python2.7/dist-packages/catkin_tools`. Is it correct? now I install this file under `share/catkin_tools`.

File: catkin_tools/commands/catkin.py [hunk 5]
```diff
@@ -178,6 +180,13 @@ def main(sysargs=None):
                     del verb_aliases[arg]
                 break
 
+    # Check for --locate-extra-shell-verbs
+    for arg in sysargs:
+        if arg == '--locate-extra-shell-verbs':
+            import inspect
+            print(os.path.abspath(os.path.join(inspect.stack()[-1][1],'../../','share/catkin_tools/catkin_shell_verbs.bash')))
```
Comment (wjwwood) [inline catkin_tools/commands/catkin.py]: This looks vestigial.

File: docs/advanced/catkin_shell_verbs.rst (hunk 4 above)
Comment (wjwwood) [inline docs/advanced/catkin_shell_verbs.rst]: I think `share/catkin_tools` is appropriate as long as you can find it easily enough at runtime. `zsh

**Comment by wjwwood** [comment]
Looks like the CI is not passing, but other than a few comments it is looking good to me.

File: docs/advanced/catkin_shell_verbs.rst (hunk 4 above)
Comment (NikolausDemmel) [inline docs/advanced/catkin_shell_verbs.rst]: Do you know a reliable way to determine where the correct `share/catkin_tools` is to be found?

Comment (k-okada) [inline docs/advanced/catkin_shell_verbs.rst]: what I found so far is follows:
```
 os.path.abspath(os.path.join(os.path.dirname(inspect.getfile(catkin_tools)), '../../../../share/catkin_tools'))
```
```
os.path.abspath(os.path.join(imp.find_module('catkin_tools')[1], '../../../../share/catkin'))
```
if we do not have other solution, we have to accept this ugly expresson, or put `catkin_shell_verbs.bash` under `dist-packages`

Comment (jbohren) [inline docs/advanced/catkin_shell_verbs.rst]: @k-okada What about this: http://peak.telecommunity.com/DevCenter/setuptools#non-package-data-files

Comment (k-okada) [inline docs/advanced/catkin_shell_verbs.rst]:
```
In [2]: resource_filename(Requirement.parse("catkin_tools"), "catkin_shell_verbs.bash")
Out[2]: '/home/k-okada/catkin_ws/ws_catkin/install/lib/python2.7/dist-packages/catkin_shell_verbs.bash'

In [3]: 
Do you really want to exit ([y]/n)? y
k-okada@kokada-t440s:~/catkin_ws/ws_catkin$ find install  -name catkin_shell_verbs.bash
install/share/catkin_tools/catkin_shell_verbs.bash
```
x-9

Comment (wjwwood) [inline docs/advanced/catkin_shell_verbs.rst]: I'd say just put it in `dist-packages` along side the code.

**Comment by wjwwood** [comment]
I fixed this up and merged it, thanks @k-okada!

**Pull request merged**